MapleServer2/PacketHandlers/Game/MatchPartyHandler.cs

```diff
@@ -0,0 +1,143 @@
+﻿using MaplePacketLib2.Tools;
+using MapleServer2.Constants;
+using MapleServer2.Enums;
+using MapleServer2.Packets;
+using MapleServer2.Servers.Game;
+using MapleServer2.Tools;
+using MapleServer2.Types;
+using Microsoft.Extensions.Logging;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MapleServer2.PacketHandlers.Game
+{
+    public class MatchPartyHandler : GamePacketHandler
+    {
+        public override RecvOp OpCode => RecvOp.MATCH_PARTY;
+
+        public MatchPartyHandler(ILogger<MatchPartyHandler> logger) : base(logger) { }
+
+        private enum MatchPartyMode : byte
+        {
+            CreateListing = 0x0,
+            RemoveListing = 0x1,
+            Refresh = 0x2
+        };
+
+        public override void Handle(GameSession session, PacketReader packet)
+        {
+            MatchPartyMode mode = (MatchPartyMode)packet.ReadByte();
+            switch (mode)
+            {
+                case MatchPartyMode.CreateListing:
+                    HandleCreateListing(session, packet);
+                    break;
+                case MatchPartyMode.RemoveListing:
+                    HandleRemoveListing(session, packet);
+                    break;
+                case MatchPartyMode.Refresh:
+                    HandleRefresh(session, packet);
+                    break;
+                default:
+                    IPacketHandler<GameSession>.LogUnknownMode(mode);
+                    break;
+            }
+        }
+
+        public void HandleCreateListing(GameSession session, PacketReader packet)
+        {
+            string partyName = packet.ReadUnicodeString();
+            bool approval = packet.ReadBool();
+            int maxMembers = packet.ReadInt();
+
+            Party party = GameServer.PartyManager.GetPartyByLeader(session.Player);
+
+            if (party == null)
+            {
+                Party newParty = new Party(GuidGenerator.Int(), GuidGenerator.Long(), maxMembers, new List<Player> { session.Player }, partyName, approval);
+                GameServer.PartyManager.AddParty(newParty);
+
+                session.Send(PartyPacket.Create(session.Player));
+                session.Send(PartyPacket.UpdateHitpoints(session.Player));
+
+                session.Player.PartyId = newParty.Id;
+                party = newParty;
+            }
+            else
+            {
+
+                party.PartyFinderId = GuidGenerator.Long();
+                party.Name = partyName;
+                party.Approval = approval;
+                party.MaxMembers = maxMembers;
+
+            }
+            party.BroadcastPacketParty(MatchPartyPacket.CreateListing(party));
+            party.BroadcastPacketParty(PartyPacket.MatchParty(party));
+            if (party.Members.Count >= maxMembers)
```

Comment:
Should the above packets even be sent in the first place if this `if` statement is true?

Reply:
Yeah, there's a weird issue where the client thinks you still have a listing open even if you never get the CreateListing packet. This is an edge case.  Only happens if you try to list and your party is already full.